int32 cmd
---
bool success
string message